controllers/managedocs_controller.go

```diff
@@ -722,6 +731,39 @@ func (r *ManagedOCSReconciler) getDesiredConvergedStorageCluster() (*ocsv1.Stora
 	return sc, nil
 }
 
+func (r *ManagedOCSReconciler) getDesiredConsumerStorageCluster() (*ocsv1.StorageCluster, error) {
+
+	capacityAsString := r.addonParams[consumerRequestedCapacity]
+	onboardingTicket := r.addonParams[consumerOnboardingTicket]
+	storageProviderEndpoint := r.addonParams[consumerStorageProviderEndpoint]
+
+	requestedCapacity, err := resource.ParseQuantity(capacityAsString)
+	if err != nil {
+		return nil, fmt.Errorf("invalid storage cluster capacity value: %v", capacityAsString)
+	}
+
+	currentState := r.storageCluster.Spec.ExternalStorage
+	sc := templates.ConsumerStorageClusterTemplate.DeepCopy()
+	es := &sc.Spec.ExternalStorage
+
+	if currentState.OnboardingTicket != onboardingTicket {
+		es.OnboardingTicket = onboardingTicket
+	}
```

Comment:
Testing that something is not equal to something then setting it is equivalent to just set it:
```
if a != A {
     a = A
}
```
is equivalent to 
```
a = A
```
while the second is simpler.

Can we please remove the if statement?